bucket/winbtrfs-np.json

```diff
@@ -29,14 +29,14 @@
             "}",
             "",
             "Write-Host \"Installing driver...\"",
-            "InfDefaultInstall \"$dir\\btrfs.inf\"",
-            "Start-Process -Wait autohotkey -ArgumentList \"$bucketsdir\\nonportable\\scripts\\$app\\no-reboot.ahk\""
+            "Invoke-ExternalCommand nefconw -ArgumentList @('--inf-default-install', '--inf-path', \"$dir\\btrfs.inf\") -RunAs -ContinueExitCodes @{ 3010 = 'A system reboot is required to finalize the uninstallation.' }"
```

Comment:
`nefconw` is a [GUI app](https://github.com/nefarius/nefcon#about). Scoop is a CLI installer. So is there any reason to use `nefconw` instead of `nefconc`, its CLI cousin? My worry is that `nefconw` might fail in situations where there isn't an attached GUI, like when SSHing into a headless server.

Reply:
> `nefconw` is a [GUI app](https://github.com/nefarius/nefcon#about). Scoop is a CLI installer. So is there any reason to use `nefconw` instead of `nefconc`, its CLI cousin? My worry is that `nefconw` might fail in situations where there isn't an attached GUI, like when SSHing into a headless server.

I use `nefconw` for a windowless execution. Most setups in this bucket are "GUI" apps in a sense, right? I don't see why it would fail, have you tried it?

Reply:
In an ssh session, I got this:
```shell
C:\>nefconw --help

C:\>nefconc --help 
usage: .\nefconc [options] [logging]

  options:
    --install-driver           Invoke the installation of a given PNP driver
      --inf-path               Absolute path to the INF file to install (required)        
    --uninstall-driver         Invoke the removal of a given PNP driver
      --inf-path               Absolute path to the INF file to uninstall (required)      
    --create-device-node       Create a new ROOT enumerated virtual device
      --hardware-id            Hardware ID of the new device (required)
      --class-name             Device Class Name of the new device (required)
      --class-guid             Device Class GUID of the new device (required)
    --remove-device-node       Removes a device and its driver
      --hardware-id            Hardware ID of the device (required)
      --class-guid             Device Class GUID of the device (required)
    --add-class-filter         Adds a service to a device class' filter collection        
      --position               Which filter to modify (required)
                                 Valid values include: upper|lower
      --service-name           The driver service name to insert (required)
      --class-guid             Device Class GUID to modify (required)
    --remove-class-filter      Removes a service to a device class' filter collection     
      --position               Which filter to modify (required)
                                 Valid values include: upper|lower
      --service-name           The driver service name to insert (required)
      --class-guid             Device Class GUID to modify (required)
    --create-driver-service    Creates a new service with a kernel driver as binary       
      --bin-path               Absolute path to the .sys file (required)
      --service-name           The driver service name to create (required)
      --display-name           The friendly name of the service (required)
    --remove-driver-service    Removes an existing kernel driver service
      --service-name           The driver service name to remove (required)
    --inf-default-install      Installs an INF file with a [DefaultInstall] section       
      --inf-path               Absolute path to the INF file to install (required)        
    --inf-default-uninstall    Uninstalls an INF file with a [DefaultUninstall] section   
      --inf-path               Absolute path to the INF file to uninstall (required)      
    -v, --version              Display version of this utility

  logging:
    --default-log-file=.\log.txt       Write details of execution to a log file (optional)
    --verbose                          Turn on verbose/diagnostic logging (optional) 
```
so `nefconw` didn't output anything to the ssh window. Neither did `nefconw -v`.

Reply:
Yeah, `nefconw` is a windowless application with the same commands as `nefconc`.

Reply:
@sitiom Since scoop is a CLI app, I think we want to use `nefconc` as it works in all environments. Is there any reason to provide `nefconw` if they both provide the same functionality, but `nefconw` fails to work in a headless CLI environment?